aea/cli/add.py

```diff
@@ -131,8 +131,8 @@ def add_item(ctx: Context, item_type: str, item_public_id: PublicId) -> None:
     if not is_fingerprint_correct(package_path, item_config):  # pragma: no cover
         raise click.ClickException("Failed to add an item with incorrect fingerprint.")
 
-    register_item(ctx, item_type, item_public_id)
     _add_item_deps(ctx, item_type, item_config)
+    register_item(ctx, item_type, item_public_id)
```

Comment:
this is a bit hacky. We first try to add the sub dependencies, and then register the current package in `aea-config.yaml`.

Consider `skill_a` with a self-loop. Up to line `134`, the package is copied in `vendor` thanks to `copy_package_directory` or `fetch_package`. When line `134` is executed, the `add` command is executed with `skill_a` as argument. However, the second time the `copy*` function will fail because it is already present.

Reply:
if approved, will open an issue to keep track of this workaround, and find a proper solution.